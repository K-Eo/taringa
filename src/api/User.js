import APIBase from './APIBase'

class User extends APIBase {
  about(username) {
    return super.execute(`/user/${username}/about`)
  }
}

export default User
