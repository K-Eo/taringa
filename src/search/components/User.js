import React from 'react'
import { Link } from 'react-router-dom'

const User = props => {
  return (
    <div className="col-6 mb-4">
      <div className="card">
        <div className="card-body p-3 d-flex">
          <img
            src={props.avatar}
            className="rounded mr-3"
            alt={props.username}
            style={{
              height: 68,
              maxHeight: 68,
              maxWidth: 68,
              minHeight: 68,
              minWidth: 68,
              width: 68,
            }}
          />
          <div>
            <h6 className="m-0">
              {props.firstname} {props.lastname}
            </h6>
            <Link to={`/u/${props.username}`} style={{ fontSize: '90%' }}>
              @{props.username}
            </Link>

            <p className="mb-0 mt-1 text-muted small">
              {props.followers} seguidores &nbsp;
              {props.following} siguiendo &nbsp;
              {props.stories} posts
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default User
