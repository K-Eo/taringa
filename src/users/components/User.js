import React from 'react'

import Feed from '../../feed/components/Feed'
import Filterable from '../../filters/components/Filterable'
import UserCard from './UserCard'

const User = ({ match }) => {
  const username = match.params.username

  const filters = {
    hot: {
      displayName: 'Destacados',
      exact: true,
      id: 'hot',
      pathname: `/u/${username}`,
      url: `/user/${username}/feed`,
    },
    recents: {
      displayName: 'Recientes',
      exact: false,
      id: 'recents',
      pathname: `/u/${username}/recents`,
      url: `/user/${username}/feed`,
    },
    tops: {
      displayName: 'Tops',
      exact: false,
      id: 'tops',
      pathname: `/u/${username}/tops`,
      url: `/user/${username}/tops/week`,
    },
  }

  const filter = match.params.filter || 'hot'
  const url = filters[filter].url

  return (
    <div className="row">
      <Filterable filters={filters} />

      <div className="col-8">
        <Feed feedId={username} filter={filter} url={url} />
      </div>

      <div className="col-4">
        <UserCard id={username} />
      </div>
    </div>
  )
}

export default User
