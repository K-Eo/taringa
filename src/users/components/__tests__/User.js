import React from 'react'

import User from '../User'

describe('User', () => {
  it('exists', () => {
    expect(User).toBeDefined()
  })

  it('renders', () => {
    const match = {
      params: {
        filter: 'hot',
        username: 'foobar',
      },
    }

    const wrapper = shallow(<User match={match} />)

    expect(wrapper).toMatchSnapshot()
  })
})
