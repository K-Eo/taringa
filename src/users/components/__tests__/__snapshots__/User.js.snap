// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders 1`] = `
<div
  className="row"
>
  <Connect(Filterable)
    filters={
      Object {
        "hot": Object {
          "displayName": "Destacados",
          "exact": true,
          "id": "hot",
          "pathname": "/u/foobar",
          "url": "/user/foobar/feed",
        },
        "recents": Object {
          "displayName": "Recientes",
          "exact": false,
          "id": "recents",
          "pathname": "/u/foobar/recents",
          "url": "/user/foobar/feed",
        },
        "tops": Object {
          "displayName": "Tops",
          "exact": false,
          "id": "tops",
          "pathname": "/u/foobar/tops",
          "url": "/user/foobar/tops/week",
        },
      }
    }
  />
  <div
    className="col-12 col-lg-8"
  >
    <Connect(FilterableInfiniteScroll)
      feedId="foobar"
      filter="hot"
      url="/user/foobar/feed"
    />
  </div>
  <div
    className="col-12 col-lg-4 order-first order-lg-last mb-4 mb-lg-0"
  >
    <Connect(UserCardContainer)
      username="foobar"
    />
  </div>
</div>
`;
